src/pcapi/core/fraud/api.py

```diff
@@ -264,23 +266,43 @@ def get_source_data(user: User) -> models.JouveContent:
     return mapped_class[fraud_check_type.type](**fraud_check_type.resultContent)
 
 
+def get_latest_fraud_result(user: User) -> typing.Optional[models.BeneficiaryFraudResult]:
+    """
+    A user's latest fraud result, if any, is the most recently updated or
+    created.
+    """
+    date_updated_col = models.BeneficiaryFraudResult.dateUpdated
+    date_created_col = models.BeneficiaryFraudResult.dateCreated
+
+    return (
+        models.BeneficiaryFraudResult.query.filter_by(userId=user.id)
+        .order_by(sa.desc(funcs.coalesce(date_updated_col, date_created_col)))
+        .first()
+    )
+
+
 def upsert_suspicious_fraud_result(user: User, reason: str) -> models.BeneficiaryFraudResult:
     """
     If the user has no fraud result: create one suspicious fraud result with
     the given reason. If it already has one: update the result's reason.
     """
-    fraud_result = models.BeneficiaryFraudResult.query.filter_by(userId=user.id).one_or_none()
+    fraud_result = get_latest_fraud_result(user)
 
     if not fraud_result:
-        fraud_result = models.BeneficiaryFraudResult(user=user, status=models.FraudStatus.SUSPICIOUS, reason=reason)
+        status = models.FraudStatus.SUSPICIOUS
+        fraud_result = models.BeneficiaryFraudResult(user=user, status=status, reason=reason)
     else:
         # if this function is called twice (or more) in a row with the same
         # reason, do not update the reason column with the same reason repeated
         # over and over. It makes the reason less readable and therefore less
         # useful.
-        last_reason = fraud_result.reason.split(FRAUD_RESULT_REASON_SEPARATOR)[-1].strip() if fraud_result else None
+        last_reason = None
+        if fraud_result:
+            last_reason = fraud_result.reason.split(FRAUD_RESULT_REASON_SEPARATOR)[-1].strip()
```

Comment:
Pourquoi ce `if fraud_result`, alors qu'on est déjà dans la branche `else` du `if not fraud_result`. Ici, on sait déjà qu'on a un `fraud_result`, non ? Je loupe quelque chose ?

```suggestion
        last_reason = fraud_result.reason.split(FRAUD_RESULT_REASON_SEPARATOR)[-1].strip()
```

Reply:
oups 😓 effectivement, ça n'a pas de sens. Je corrige !